interactive_engine/executor/ir/graph_proxy/src/adapters/gs_store/read_graph.rs

```diff
@@ -524,6 +549,154 @@ fn edge_trim(mut edge: Edge, columns: Option<&Vec<NameOrId>>) -> Edge {
     edge
 }
 
+pub struct HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    lazy: AtomicPtr<V>,
+    cached: Option<HashMap<PropId, Object>>,
+}
+
+impl<V> HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    pub fn new(v: V) -> Self {
+        let ptr = Box::into_raw(Box::new(v));
+        HybridVertexDetails { lazy: AtomicPtr::new(ptr), cached: None }
+    }
+
+    fn get_vertex_ptr(&self) -> Option<*mut V> {
+        let ptr = self.lazy.load(Ordering::SeqCst);
+        if ptr.is_null() {
+            None
+        } else {
+            Some(ptr)
+        }
+    }
+}
+
+impl<V> fmt::Debug for HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("gs_store LazyVertexDetails")
+            .field("lazy", &self.lazy)
+            .field("cached", &self.cached)
+            .finish()
+    }
+}
+
+impl<V> AsAny for HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+
+    fn as_any_ref(&self) -> &dyn Any {
+        self
+    }
+}
+
+impl<V> Drop for HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    fn drop(&mut self) {
+        let ptr = self.lazy.load(Ordering::SeqCst);
+        if !ptr.is_null() {
+            unsafe {
+                std::ptr::drop_in_place(ptr);
+            }
+        }
+    }
+}
+
+impl<V> Details for HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    fn get_property(&self, key: &NameOrId) -> Option<PropertyValue> {
+        if let NameOrId::Id(key) = key {
+            if let Some(ref cached) = self.cached {
+                cached
```

Comment:
This may not be right. It is possible that we cached some properties, but not including the queried one.